### PR TITLE
Data model

### PR DESCRIPTION
**Pull request by ridabensalem**
in thus milestone we have added the following requirements : 

- Create a new app. Give it a meaningful name.
- Initialize your project with Git.
- Make sure that your project has Postgres database set up.
- Build your project schema.
- Create and run the necessary migration files.
- Table and column names should match the ERD diagram.
- note: photo for users table should be a link to an image
- Foreign keys should be included.
- Pay attention that in posts table the column holding the foreign key to the users table should be named author_id
- All columns that are foreign keys should have a corresponding index.

### CODE REVIEW

File: db/schema.sql [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  updated_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  commentsCounter INTEGER NOT NULL DEFAULT 0,
+    likesCounter INTEGER NOT NULL DEFAULT 0,
+    foreign key (id) references comment(post_id) on delete cascade,
+    foreign key (id) references likes(post_id) on delete cascade
+);
+--like shema
+CREATE TABLE likes (
+  post_id SERIAL PRIMARY KEY,
+  user_id SERIAL PRIMARY KEY,
```
Comment (omar-muhamad) [inline db/schema.sql:38]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/creating_data_model_project_v1_2.md#:~:text=your%20project%20schema.-,Create%20and%20run%20the%20necessary%20migration%20files.,-Table%20and%20column), Please create all necessary migration files here in `./db/migrate` regarding tables name, foreign keys, and columns included on the tables.
[Hint]: You suppose to use `rails generate migration` command for that, See this [link](https://guides.rubyonrails.org/active_record_migrations.html#creating-a-migration) for more info.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @ridabensalem,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: db/migrate/20230711111829_create_comments.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreateComments < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+      t.text :text, null: false
+      t.references :user, foreign_key: { on_delete: :cascade }
+      t.references :post, foreign_key: { on_delete: :cascade }
+      t.timestamps
+    end
```
Comment (Davidosky007) [inline db/migrate/20230711111829_create_comments.rb]: * Kindly ensure Table and column names should match the ERD diagram as stated in the [requirement ](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/creating_data_model_project_v1_2.md#:~:text=Table%20and%20column%20names%20should%20match%20the%20ERD%20diagram.)
image

File: db/migrate/20230711112658_create_likes.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class CreateLikes < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes do |t|
+      t.references :post, null: false, foreign_key: true
+      t.references :user, null: false, foreign_key: true
+      t.datetime :created_at
+      t.datetime :updated_at
+
+      t.timestamps
+    end
```
Comment (Davidosky007) [inline db/migrate/20230711112658_create_likes.rb]: * Kindly ensure Table and column names should match the ERD diagram as stated in the [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/creating_data_model_project_v1_2.md#:~:text=Table%20and%20column%20names%20should%20match%20the%20ERD%20diagram.)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @ridabensalem 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* Migration files used to maintain database 👍 
* Well written PR 👍 
* Good looking readme 👍 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omarsalem7 — APPROVED** [review]
Hi @ridabensalem 😊, 
# STATUS: APPROVED! 🥇 💯 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**